### PR TITLE
Add ktlint and formatting the code

### PR DESCRIPTION
**Pull request by Airsaid**
Add ktlint and formatting the code.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -47,4 +48,10 @@ dependencies {
   androidTestImplementation(libs.androidx.junit)
   androidTestImplementation(libs.androidx.espresso)
   implementation(projects.extension)
-}
\ No newline at end of file
+}
+
+ktlint {
```
Comment (JackChen365) [inline app/build.gradle.kts]: Is it better to configure it this way? So all the sub modules could apply the ktlint

```
allprojects { project->
    project.plugins.apply("org.jlleitschuh.gradle.ktlint")
    project.tasks.withType(org.jlleitschuh.gradle.ktlint.tasks.BaseKtLintCheckTask).configureEach {
        it.workerMaxHeapSize.set("512m")
    }
    project.ktlint {
        version = "0.45.2"
        debug = false
        verbose = true
        android = true
        outputToConsole = true
        outputColorName = "RED"
        ignoreFailures = false
        enableExperimentalRules = true
        disabledRules = ["final-newline","import-ordering","spacing-around-angle-brackets"]
        reporters { reporter "html" }
        baseline = file("config/ktlint/baseline.xml")
        filter {
            exclude("**/generated/**")
            include("**/kotlin/**")
        }
    }
}
```

File: app/build.gradle.kts [hunk 2]
```diff
@@ -2,6 +2,7 @@ plugins {
   id("com.android.application")
   id("org.jetbrains.kotlin.android")
   id("io.github.jackchen365.sample")
+  id("org.jlleitschuh.gradle.ktlint")
```
Comment (JackChen365) [inline app/build.gradle.kts]: As I mentioned, We better add this `org.jlleitschuh.gradle.ktlint-idea`.

Therefore, We can have those two tasks.

image

**Review by JackChen365 — APPROVED** [review]
LGTM

File: app/build.gradle.kts (hunk 2 above)
Comment (Airsaid) [reply app/build.gradle.kts]: In fact, the ktlint core plugin already apply the plugin:
```kotiln
open class KtlintPlugin : Plugin<Project> {
    override fun apply(target: Project) {
        // Apply the idea plugin
        target.plugins.apply(KtlintIdeaPlugin::class.java)
        ......
    }
    ......
}
```

Comment (Airsaid) [reply app/build.gradle.kts]: But, if we need to enable add git hook for these tasks, we need to apply it in the root project.

File: app/build.gradle.kts (hunk 1 above)
Comment (Airsaid) [reply app/build.gradle.kts]: Done

**Pull request merged**